### DISCUSSION TITLE
understanding-security-headers-part-2-cross-origin-resource-policy-preventing-hotlinking/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#442] [Announcements]
# understanding-security-headers-part-2-cross-origin-resource-policy-preventing-hotlinking/

In this post I describe how to use the Cross-Origin-Resource-Policy to block cross-origin requests that would normally be allowed, such as in <img> tags

https://andrewlock.net/understanding-security-headers-part-2-cross-origin-resource-policy-preventing-hotlinking/

**Comment by kapsiR** [comment]
> As you can see in the above image, in this scenario the CORS header is all that's required; no CORP header is applied and it does not affect the loading of the image.

It would be more obvious if the example has a CORP header of `same-origin`, what do you think?

**Comment by RichardD2** [comment]
> This is often called [hotlinking](https://simple.wikipedia.org/wiki/Hotlinking) and is problematic as it makes it easier to plagiarise content, but also places the egress and server costs on your site, as the host, instead of the embedding site.
> ...  
> If we take a look at the dev tools, we can see that the request was made successfully, and returned a 200 OK response, but the browser rejected the content, because it was a cross-origin request with a same-origin CORP header:

Perhaps I'm missing something, but that seems to suggest you *still* face the egress and server costs. Your server still serves the image, but the browser then rejects it.
